### PR TITLE
FIXED : Make the Sphinx modules headers match the filenames

### LINKED ISSUE #17493: Make the Sphinx modules headers match the filenames
Right now it's hard to navigate the "SymPy Modules Reference" subheaders at https://docs.sympy.org/latest/index.html, because the names are not in alphabetical order.  There are two issues here:

- The header names are inconsistent. For example, things like "SymPy Core" should just be "Core". It would be best if each header was just the name of the submodule.

- The headers are not in order. They are listed in this file https://github.com/sympy/sympy/blob/master/doc/src/modules/index.rst. 

CC @moorepants @certik @lglattly

### PR DESCRIPTION
**Pull request by harsh9200**
#### References to other Issues or PRs

**Fixes #17493** 

#### Brief description of what is fixed or changed
Navigating  the subheaders  of "SymPy Modules Reference" at https://docs.sympy.org/latest/index.html in alphabetical order and  fixing the  following isssues:

-  The header names are inconsistent. For example, things like "SymPy Core" should just be 

 "Core"
- The headers are not in alphabetical order.

#### Release Notes

- other

   - Edited all the header names of the doc file to match the name of submodule.

    - Sorted the Index in alphabetical order.

### CODE REVIEW

File: doc/src/modules/algebras.rst [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ===============
-Algebras Module
+Algebras
 ===============
```
Comment (oscarbenjamin) [inline doc/src/modules/algebras.rst]: The number of `===` should match the length of the title.

File: doc/src/modules/evalf.rst [hunk 2]
```diff
@@ -1,6 +1,6 @@
 .. _evalf-label:
 
-Numerical evaluation
+Numerical Evaluation
 ====================
```
Comment (oscarbenjamin) [inline doc/src/modules/evalf.rst]: Some of these are not really modules and should probably be listed as subtopics under another heading. This one could go under core for example.

File: doc/src/modules/functions/index.rst [hunk 3]
```diff
@@ -1,6 +1,6 @@
 .. _functions:
 
-Functions Module
+Functions
 ****************
```
Comment (oscarbenjamin) [inline doc/src/modules/functions/index.rst]: Why is this using `***` instead of `===`?

File: doc/src/modules/index.rst [hunk 4]
```diff
@@ -16,49 +16,49 @@ access any SymPy module, or use the list below:
...
    sets.rst
    simplify/simplify.rst
-   simplify/hyperexpand.rst
-   stats.rst
-   solvers/ode.rst
-   solvers/pde.rst
    solvers/solvers.rst
-   solvers/diophantine.rst
-   solvers/inequalities.rst
    solvers/solveset.rst
```
Comment (oscarbenjamin) [inline doc/src/modules/index.rst]: These could all be under a solvers header. Perhaps solvers.rst could be the main header there

File: doc/src/modules/index.rst [hunk 5]
```diff
@@ -16,49 +16,49 @@ access any SymPy module, or use the list below:
...
+   stats.rst
    tensor/index.rst
+   rewriting.rst
```
Comment (oscarbenjamin) [inline doc/src/modules/index.rst]: This isn't really a module and probably belongs under core

File: doc/src/modules/integrals/integrals.rst [hunk 6]
```diff
@@ -129,7 +129,7 @@ any order and any precision:
 
 .. autofunction:: sympy.integrals.quadrature.gauss_lobatto
 
-Integration over Polytopes
+Integration Over Polytopes
```
Comment (oscarbenjamin) [inline doc/src/modules/integrals/integrals.rst]: I wouldn't capitalise the work "over" here.

File: doc/src/modules/polys/index.rst [hunk 7]
```diff
@@ -1,7 +1,7 @@
 .. _polys-docs:
 
 ===============================
-Polynomials Manipulation Module
+Polynomials Manipulation
 ===============================
```
Comment (oscarbenjamin) [inline doc/src/modules/polys/index.rst]: The module is called "polys" but if I were writing a title like above I would use the singular "polynomial manipulation".

File: doc/src/modules/index.rst (hunk 4 above)
Comment (harsh9200) [reply doc/src/modules/index.rst]: so i place all other files code in solvers.rst?

Comment (oscarbenjamin) [reply doc/src/modules/index.rst]: No. I mean that there could be a header for solvers with the other things underneath like:
```
   * solvers
        * diophantine
        * inequalities
        ...
```
I'm not sure how to do that in the RST though...

Comment (harsh9200) [reply doc/src/modules/index.rst]: okay let me try

**Comment by sylee957** [comment]
Module names would literally be corresponding to how python imports be happening

Category Theory -> Categories
Code Generation -> Codegen
Cryptography -> Crypto
Differential Geometry -> Diffgeom
Polynomial Manipulation -> Polys

But I think that it's okay to have names like left, than using abbreviations like right, and also it would already not be very consistent of having `ODE` or `Diophantine` exposed at the top level, since they should be submodules for the solvers.

I wonder if there are other opinions about this.

File: doc/src/modules/index.rst (hunk 4 above)
Comment (harsh9200) [reply doc/src/modules/index.rst]: done added all the solvers file in one heading and the rewriting.rst and evalf.rst to core

**Comment by oscarbenjamin** [comment]
This looks good. I like the way that under solvers there is a page with a more detailed TOC that has:
```
* Solvers
   * Algebraic equations
   * Ordinary Differential equations (ODEs)
   * Partial Differential Equations (PDEs)
   * Deutils (Utilities for solving ODE’s and PDE’s)
   * Recurrence Equations
   * Systems of Polynomial Equations
   * Diophantine Equations (DEs)
   * Inequalities
* Solveset
   * What’s wrong with solve():
   * Why Solveset?
...
```
However I would like to see the top-level headings here listed in the main TOC on the front page of the docs e.g.:
```
* Simplify
* Solvers
    * Solvers
    * Solveset
    ...
* Stats
```
Some of the topics in the main TOC are really subtopics of each other so I suggest they should be nested. I don't know if this is something you can directly control though.

**Comment by harsh9200** [comment]
> This looks good. I like the way that under solvers there is a page with a more detailed TOC that has:
> 
> ```
> * Solvers
>    * Algebraic equations
>    * Ordinary Differential equations (ODEs)
>    * Partial Differential Equations (PDEs)
>    * Deutils (Utilities for solving ODE’s and PDE’s)
>    * Recurrence Equations
>    * Systems of Polynomial Equations
>    * Diophantine Equations (DEs)
>    * Inequalities
> * Solveset
>    * What’s wrong with solve():
>    * Why Solveset?
> ...
> ```
> 
> However I would like to see the top-level headings here listed in the main TOC on the front page of the docs e.g.:
> 
> ```
> * Simplify
> * Solvers
>     * Solvers
>     * Solveset
>     ...
> * Stats
> ```
> 
> Some of the topics in the main TOC are really subtopics of each other so I suggest they should be nested. I don't know if this is something you can directly control though.

can you explain a little more?

**Comment by oscarbenjamin** [comment]
I mean that rather than removing topics from the main TOC you could organise them into topics with subtopics so that e.g. ODE is a subtopic of solvers.

**Comment by harsh9200** [comment]
> I mean that rather than removing topics from the main TOC you could organise them into topics with subtopics so that e.g. ODE is a subtopic of solvers.

i am trying but i am unable to find anything on google

**Comment by oscarbenjamin** [comment]
I also don't know how to do it so maybe it's best not to bother. Can you make sure that the main TOC still has the same sections that it did before nut just renamed and reordered?

**Comment by harsh9200** [comment]
> I also don't know how to do it so maybe it's best not to bother. Can you make sure that the main TOC still has the same sections that it did before nut just renamed and reordered?

all done?

**Comment by oscarbenjamin** [comment]
I think this looks good. @sylee957 does it look good to you?

File: doc/src/modules/plotting.rst [hunk 8]
```diff
@@ -87,7 +88,7 @@ Series Classes
 
 
 
-Pyglet Plotting Module
+Pyglet Plotting
 ======================
```
Comment (sylee957) [inline doc/src/modules/plotting.rst]:
```suggestion
---------------
```
This header should be changed to make the TOC tree correct.

**Comment by oscarbenjamin** [comment]
Thanks for this. I've merged it.

**Pull request merged**